data/transactions/logic/doc.go

```diff
@@ -286,13 +286,15 @@ var opDescByName = map[string]OpDesc{
 	"frame_bury": {"replace the Nth (signed) value from the frame pointer in the stack with A", "", []string{"frame slot"}},
 	"popn":       {"remove N values from the top of the stack", "", []string{"stack depth"}},
 
-	"box_create":  {"create a box named A, of length B. Fail if A is empty or B exceeds 32,768. Returns 0 if A already existed, else 1", "Newly created boxes are filled with 0 bytes. `box_create` will fail if the referenced box already exists with a different size. Otherwise, existing boxes are unchanged by `box_create`.", nil},
+	"box_create":  {"create a box named A, of length B. Fail if the name A is empty or B exceeds 32,768. Returns 0 if A already existed, else 1", "Newly created boxes are filled with 0 bytes. `box_create` will fail if the referenced box already exists with a different size. Otherwise, existing boxes are unchanged by `box_create`.", nil},
 	"box_extract": {"read C bytes from box A, starting at offset B. Fail if A does not exist, or the byte range is outside A's size.", "", nil},
 	"box_replace": {"write byte-array C into box A, starting at offset B. Fail if A does not exist, or the byte range is outside A's size.", "", nil},
+	"box_splice":  {"set box A to contain its previous bytes up to index B, followed by D, followed by the original bytes of A that began at index B+C.", "Boxes are of constant length. If C < len(D), then len(D)-C bytes will be removed from the end. If C > len(D), zero bytes will be appended to the end to reach the box length.", nil},
 	"box_del":     {"delete box named A if it exists. Return 1 if A existed, 0 otherwise", "", nil},
 	"box_len":     {"X is the length of box A if A exists, else 0. Y is 1 if A exists, else 0.", "", nil},
 	"box_get":     {"X is the contents of box A if A exists, else ''. Y is 1 if A exists, else 0.", "For boxes that exceed 4,096 bytes, consider `box_create`, `box_extract`, and `box_replace`", nil},
 	"box_put":     {"replaces the contents of box A with byte-array B. Fails if A exists and len(B) != len(box A). Creates A if it does not exist", "For boxes that exceed 4,096 bytes, consider `box_create`, `box_extract`, and `box_replace`", nil},
+	"box_resize":  {"change the size of box named A to be of length B, adding zero bytes to end or removing bytes from the end, as needed. Fail if the name A is empty, A is not an existing box, or B exceeds 32,768.", "", nil},
```

Comment:
Do you think it's worth mentioning or alluding to the fact that the MBR will change if the box is resized?

I don't feel strongly about this because no other opcodes seem to mention MBR changes as a consequence, so maybe this isn't the place for it.

Reply:
Chiming in to say I'd mention it, just because it is one of the reasons why we plan to use this opcode: `box_resize` is what we need for a feature where, not only to resize the box dynamically avoiding big MBR pre-allocation, but also because we need to return the MBR to who expanded the box.

Reply:
I'll try to figure out a way to add it.  I'm not sure if it should be here, since we don't describe MBR changes here for anything else, as @jasonpaulos pointed out. But we must have some place where it makes sense to say it.

Maybe I can make sure that discussion appears elsewhere and just annotate every opcode that can change the MBR with "May affect MBR" or similar.

Reply:
I added a note in the main README about MBR changing immediately, and failure if the account doesn't have the algos.